### PR TITLE
Implemented Undefined class

### PR DESCRIPTION
**Pull request by badochov**

### CODE REVIEW

File: src/runtime/undefined.go [hunk 1]
```diff
@@ -1,10 +1,294 @@
...
+}
+
+func (u Undefined) undefinedMessage() string {
+	if u.hint != nil {
+		return *u.hint
+	}
+	if _, ok := u.obj.(utils.Missing); ok {
+		return fmt.Sprintf("'%s' is undefined", *u.name)
+	}
+	// Following if is a rewrite of python code below. I don't undeestand neither the message nor the logic, as undefined_name ought to be Optional[string].
```
Comment (jmolinski) [inline src/runtime/undefined.go:64]: Let's hope it's a mistake in jinja and that undefined_name shouldn't in fact be `Any`

File: src/runtime/undefined.go [hunk 2]
```diff
@@ -1,10 +1,294 @@
...
+
+func (su StrictUndefined) Contains(any) (bool, error) {
+	return false, su.failWithUndefinedError()
+}
+
+func objectTypeRepr(a any) string {
+	return reflect.TypeOf(a).Name()
+}
+
+func (du DebugUndefined) String_() (string, error) {
```
Comment (jmolinski) [inline src/runtime/undefined.go:265]: `String_()`?

Comment (badochov) [reply src/runtime/undefined.go:265]: I had to distinguish from `func String() string` that's used, eg by fmt.Sprint

**Review by jmolinski — APPROVED** [review]
LGTM

**Pull request merged**